test/misc.jl

```diff
@@ -842,3 +842,15 @@ end
 @testset "fieldtypes Module" begin
     @test fieldtypes(Module) isa Tuple
 end
+
+# Test that read fault on a prot-none region does not incorrectly give
+# ReadOnlyMemoryEror, but rather crashes the program
+const MAP_ANONYMOUS_PRIVATE = Sys.isbsd() ? 0x1002 : 0x22
+let script = :(let ptr = Ptr{Cint}(ccall(:jl_mmap, Ptr{Cvoid},
```

Comment:
Do we actually guarantee this everywhere else?

Reply:
Yes, I think we have this logic on every platform.

Reply:
It throws readonlymmoryerror on linux for example. I don't really see why it's that important to differentiate reading `PROT_NONE` from writing `PROT_READ`. The linux one only check for permission violation and would tread violation of read/write/execution the same.

Reply:
I imaging what you could do is to unmap the page before issueing the `unsafe_load`. Assuming the code is compiled it should make sure you are loading from a hole in the mapping and that should be a crash.

Reply:
> I imaging what you could do is to unmap the page before issueing the `unsafe_load`. Assuming the code is compiled it should make sure you are loading from a hole in the mapping and that should be a crash.

That doesn't exercise this code path. We explicitly check that the mapping exists first.

Reply:
I think we may just want to fix the linux behavior here to bring it in line with mac/linux. I don't see a good reason to throw ReadOnlyMemoryError here.

Reply:
... Sure, that's fine. It's technically breaking but it was an error before anyway...

Reply:
Or it's a bug fix and brings linux in line with the other platforms ;).

Reply:
Breaking for ppl (like me) who hardly ever use julia anywhere other than linux... Not that I've ever cought any read only memory error when reading things anyway.............

Reply:
I do agree the name of the exception is very suggestive of what it **should** do...